src/sessions.ts

```diff
@@ -286,12 +288,12 @@ export class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
    */
   startTransaction(options?: TransactionOptions): void {
     if (this[kSnapshotEnabled]) {
-      throw new MongoDriverError('Transactions are not allowed with snapshot sessions');
+      throw new MongoTransactionError('Transactions are not allowed with snapshot sessions');
```

Comment:
this is more of an API error - probably the compatibility one, because in order to get here, the user needs to intentionally start a snapshot session and then try to start a transaction on it